kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaTopicConsumerManager.java

```diff
@@ -237,6 +237,9 @@ public void close() {
         }
 
         return MessageMetadataUtils.asyncFindPosition(ledger, offset, skipMessagesWithoutIndex).thenApply(position -> {
+            if (position == null) {
+                return null;
+            }
```

Comment:
This change LGTM because `ManagedLedgerImpl#asyncFindPosition` might return a null completed future. But since this future will be put into `KafkaTopicConsumerManager#cursors`, could you double check if the null check is performed when processing the future's result from `cursors`?

Reply:
I checked all the places where `cursors` was used, all path has null check.